autotst/reaction.py

```diff
@@ -311,49 +312,64 @@ def get_labeled_reaction(self):
         assert (
             self.label or self.rmg_reaction), "You must provide a reaction or a reaction label"
 
-        match = False
-        if self.label:
-            rmg_reactants = []
-            rmg_products = []
-            r, p = self.label.split("_")
-            for react in r.split("+"):
-                s = RMGMolecule(smiles=react)
-                rmg_reactants.append(s)
-            for prod in p.split("+"):
-                s = RMGMolecule(smiles=prod)
-                rmg_products.append(s)
-
-            test_reaction = RMGReaction(
-                reactants=rmg_reactants, 
-                products=rmg_products)
-
-            if self.rmg_reaction:
-                assert self.rmg_reaction.is_isomorphic(
-                    test_reaction), "The reaction label provided does not match the RMGReaction provided..."
-
+        match = False # We have not found a labeled reaction that matches the one provided.
+        if self.label: # A reaction label was provided
+            
+            # Generating lists of lists. Main list is the reactans or products
+            # Secondary list is composed of the resonance structures for that species
+            r, p = self.label.split("_") 
+            rmg_reactants = [RMGMolecule(smiles=smile).generate_resonance_structures() for smile in r.split("+")]
+            rmg_products = [RMGMolecule(smiles=smile).generate_resonance_structures() for smile in p.split("+")]
+
+            combos_to_try = list(itertools.product(
+                itertools.product(*rmg_reactants),
+                itertools.product(*rmg_products)
+            ))
+
+            # looping though each reaction family and each combination of reactants and products
             for name, family in list(self.rmg_database.kinetics.families.items()):
-                if match:
-                    break
-                try:
-                    labeled_r, labeled_p = family.get_labeled_reactants_and_products(
-                        test_reaction.reactants, test_reaction.products)
-                except ValueError:
-                    continue
-                    
-                if not (labeled_r and labeled_p):
-                    continue
+                logging.info("Trying to match reacction to {}".format(family))
+                for rmg_reactants, rmg_products in combos_to_try:
+                    # Making a test reaction
+                    test_reaction = RMGReaction(
+                        reactants=list(rmg_reactants), 
```

Comment:
then you wouldn't need to here.
and, furthermore, if you end up using the same iterator more than once, it won't be empty the second time. (not sure if that happens)

Reply:
HOWEVER, `rmg_reactants` and `rmg_products` still need to turned into a list when generating the `test_reaction`. At this point `rmg_reactants` and `rmg_products` are tuples and an `RMGReaction` only accepts lists...